### PR TITLE
Update controller.rst clarifying automatic deletion for flash messages

### PR DESCRIPTION
**Pull request by miguelvilata**
Hello. 

It's neccesary clarify that flash messages aren't removed automatically when next session comes. They are removed because the use of  get() method in FlashBag object. If you use peek() method, or simply don't show stored messages they remain in session.

Show get() method in Symfony\Component\HttpFoundation\Session\Flash\FlashBag:

```
/**
 * {@inheritdoc}
 */
public function get($type, array $default = array())
{
    if (!$this->has($type)) {
        return $default;
    }

    $return = $this->flashes[$type];

    unset($this->flashes[$type]);

    return $return;
}
```

### CODE REVIEW

File: book/controller.rst [hunk 1]
```diff
@@ -619,8 +619,7 @@ session.
 Flash Messages
 ~~~~~~~~~~~~~~
 
-You can also store small messages that will be stored on the user's session
-for exactly one additional request. This is useful when processing a form:
+You can also store small messages that will be stored on the user's session. This is useful when processing a form:
```
Comment (xabbuh) [inline book/controller.rst]: Can you please revert this change? We wrap lines after the first word that crosses the 72nd character mark.

File: book/controller.rst [hunk 2]
```diff
@@ -675,9 +674,9 @@ the ``notice`` message:
             </div>
         <?php endforeach ?>
 
-By design, flash messages are meant to live for exactly one request (they're
-"gone in a flash"). They're designed to be used across redirects exactly as
-you've done in this example.
+.. tip::
+
+    Please note that using get() method in session :class:`Symfony\\Component\\HttpFoundation\\Session\\Flash\\FlashBag` object, automatically remove messages from session. If you want preserve flash messages, consider use peek() method instead.
```
Comment (xabbuh) [inline book/controller.rst]: I think we should keep the existing paragraph and just reword it a bit. We can then add a tip for the `peek()` method like this:

```rst
By design, flash messages are meant to be processed exactly once. This means
that they vanish from the session automatically when they are retrieved from
the flash bag by calling the ``get()`` method.

.. tip::

    You can use the
    :method:`Symfony\\Component\\HttpFoundation\\Session\\Flash\\FlashBagInterface::peek()`
    method instead to retrieve the message while keeping it in the bag.
```

What do you think?

Comment (miguelvilata) [inline book/controller.rst]: I prefer your proposition. It's more clear.

File: book/controller.rst (hunk 1 above)
Comment (miguelvilata) [inline book/controller.rst]: Ok. But what do you think about removing or modify "for exactly one additional request" sentence?. Do you think adding the tip about get()/peek() will be enough without modifiying this sentence?.

File: book/controller.rst [hunk 3]
```diff
@@ -619,8 +619,8 @@ session.
 Flash Messages
 ~~~~~~~~~~~~~~
 
-You can also store small messages that will be stored on the user's session
-for exactly one additional request. This is useful when processing a form:
+You can also store small messages that will be stored on the user's session. 
+This is useful when processing a form:
```
Comment (xabbuh) [inline book/controller.rst]: looks like something went wrong here (this doesn't match the following line)

**Comment by xabbuh** [comment]
Hey @miguelvilata, can you make the change? :)

**Comment by miguelvilata** [comment]
There is something more needed here?

**Comment by xabbuh** [comment]
:+1: looks great

**Comment by xabbuh** [comment]
Thank you @miguelvilata. I have merged your changes into the `2.3` branch (that's why your pull request is displayed as closed) and made some minor tweaks in 917e6ca.

**Comment by miguelvilata** [comment]
Thank you for your help.

__________________________
Miguel Vilata
@miguelvilata
miguel.vilata@gmail.com <wchemi@gmail.com>

2015-10-03 11:25 GMT+02:00 Christian Flothmann <notifications@github.com>:

**Comment by xabbuh** [comment]
You're welcome. Hope to see you back soon with more changes. :)
